packages/binding-http/test/http-client-basic-test.ts

```diff
@@ -110,9 +110,11 @@ class HttpClientBasicTest {
 
         this.client.setSecurity([scheme], { username: "other", password: "other" });
         try {
-            await this.client.readResource({ href: "https://localhost:3001" });
+            await this.client.readResource({ href: "https://127.0.0.1:3001" });
         } catch (error) {
-            error.message.should.eql("Client error: Unauthorized");
+            // Note: depending on Node.js version different errors appear
+            // AssertionError: expected 'request to https://127.0.0.1:3001/ fa…' to deeply equal 'Client error: Unauthorized'
+            // error.message.should.eql("Client error: Unauthorized");
```

Comment:
Mmm there's something odd here. The error should be coming from `checkFetchResponse`, so it seems that with some NodeJS version that code is being bypassed. I would like to investigate more on this. Do you remember which NodeJS version had a different error response?

Reply:
In the versions up to 16 the *old* code worked.
This means
* Node.js 12, 14  and 16 reported "Client error: Unauthorized" 
* Node.js 18 reports 'request to https://127.0.0.1:3001/ fa…'

Reply:
just tested with node 18.4.0 and I got the Client error: Unauthorized error. Was it ok windows or ubuntu?

Reply:
Since I am not using Node.JS 18 on my machine I looked at the log that the CI build caused to fail.
See first commit of this PR failed with https://github.com/eclipse/thingweb.node-wot/actions/runs/3060538348/jobs/4939172916
It failed on Ubuntu...

Reply:
Are you talking about this line? https://github.com/eclipse/thingweb.node-wot/actions/runs/3060538348/jobs/4939172916#step:7:309

Reply:
Exactly

Reply:
The error is: 
```
 https://127.0.0.1:3001/ failed, reason: connect ECONNREFUSED 127.0.0.1:3001
```
which should not happen and the test should fail. We have to understand why server is not responding in ubuntu

Reply:
This is indeed strange since the previous test "should authorize client with basic" succeeds and works... Hence the connection is not refused..

Reply:
I suggest to tackle this in a follow-up issue or @relu91  do you want to fix it in this PR?